subsys/bluetooth/audio/vocs_client.c

```diff
@@ -0,0 +1,688 @@
+		BT_DBG("Discovery complete for VOCS %p", inst);
+		inst->cli.busy = false;
+		(void)memset(params, 0, sizeof(*params));
+
+		if (inst->cli.cb && inst->cli.cb->discover) {
+			int err = valid_inst_discovered(inst) ? 0 : -ENOENT;
+
+			inst->cli.cb->discover(conn, inst, err);
+		}
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	BT_DBG("[ATTRIBUTE] handle 0x%04X", attr->handle);
+
+	if (params->type == BT_GATT_DISCOVER_CHARACTERISTIC) {
+		struct bt_gatt_subscribe_params *sub_params = NULL;
+		struct bt_gatt_chrc *chrc;
+
+		chrc = (struct bt_gatt_chrc *)attr->user_data;
+		if (inst->cli.start_handle == 0) {
+			inst->cli.start_handle = chrc->value_handle;
+		}
+		inst->cli.end_handle = chrc->value_handle;
+
+		if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_STATE)) {
+			BT_DBG("Volume offset state");
+			inst->cli.state_handle = chrc->value_handle;
+			sub_params = &inst->cli.state_sub_params;
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_LOCATION)) {
+			BT_DBG("Location");
+			inst->cli.location_handle = chrc->value_handle;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				sub_params = &inst->cli.location_sub_params;
+			}
+			if (chrc->properties & BT_GATT_CHRC_WRITE_WITHOUT_RESP) {
+				inst->cli.location_writable = true;
+			}
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_CONTROL)) {
+			BT_DBG("Control point");
+			inst->cli.control_handle = chrc->value_handle;
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_DESCRIPTION)) {
+			BT_DBG("Description");
+			inst->cli.desc_handle = chrc->value_handle;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				sub_params = &inst->cli.desc_sub_params;
+			}
+			if (chrc->properties & BT_GATT_CHRC_WRITE_WITHOUT_RESP) {
+				inst->cli.desc_writable = true;
+			}
+		}
+
+		if (sub_params) {
+			int err;
+
+			sub_params->value = BT_GATT_CCC_NOTIFY;
+			sub_params->value_handle = chrc->value_handle;
+			/*
+			 * TODO: Don't assume that CCC is at handle + 2;
+			 * do proper discovery;
+			 */
+			sub_params->ccc_handle = attr->handle + 2;
+			sub_params->notify = vocs_client_notify_handler;
+			err = bt_gatt_subscribe(conn, sub_params);
+
+			if (err) {
+				BT_WARN("Could not subscribe to handle %u",
+					sub_params->ccc_handle);
+			}
+		}
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+int bt_vocs_client_state_get(struct bt_conn *conn, struct bt_vocs *inst)
+{
+	int err;
+
+	CHECKIF(!conn) {
+		BT_DBG("NULL conn");
+		return -ENOTCONN;
+	}
+
+	CHECKIF(!inst) {
+		BT_DBG("NULL instance");
+		return -EINVAL;
+	}
+
+	if (!inst->cli.state_handle) {
+		BT_DBG("Handle not set");
+		return -EINVAL;
+	}
+
+	if (inst->cli.busy) {
+		BT_DBG("Handle not set");
+		return -EBUSY;
+	}
+
+	inst->cli.read_params.func = vocs_client_read_offset_state_cb;
+	inst->cli.read_params.handle_count = 1;
+	inst->cli.read_params.single.handle = inst->cli.state_handle;
+	inst->cli.read_params.single.offset = 0U;
+
+	err = bt_gatt_read(conn, &inst->cli.read_params);
+	if (!err) {
+		inst->cli.busy = true;
+	}
+	return err;
```

Comment:
Here I wouldn't necessarily complain about the missing empty line after `}` since it's kind of a singular chunk of code, but you're actually not being consistent with yourself: there are other places such as the end of `bt_vocs_client_location_get()`, `bt_vocs_client_state_set()` and `bt_vocs_client_description_get()` where you *do* have the empty line, so I'd suggest to go with the same style here.

Reply:
Agreed; tried to update it after your last comment regarding this. Must've missed this one :)